pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -472,6 +520,30 @@ func fixCrcDnsmasqConfigFile() error {
 	return nil
 }
 
+func removeCrcDnsmasqConfigFile() error {
+	if err := checkNetworkManagerInstalled(); err != nil {
+		// Since NetworkManager is not installed then this file will not present.
+		return nil
+	}
+	if _, err := os.Stat(crcDnsmasqConfigPath); !os.IsNotExist(err) {
```

Comment:
In my opinion, if we get an error, and it's not a "file does not exist" error, we should return it right away rather than trying to remove the dnsmasq config (I think this is what would happen for now?)
```
_, err := os.Stat(crcDnsmasqConfigPath)
if err != nil {
    if os.IsNotExist(err) {
        return nil
    } else {
        return err
    }
}
logging.Debug("Removing dnsmasq configuration")
````

Reply:
@cfergeau As per go docs https://golang.org/pkg/os/#File.Stat is there is an error it is only the PathError https://golang.org/pkg/os/#PathError this logic is used a lot of place without any issue https://gist.github.com/mattes/d13e273314c3b3ade33f

Reply:
Yes but `PathError` is not always a "file not  found" error as indicated in https://gist.github.com/mattes/d13e273314c3b3ade33f it could for example match `os.IsPermission()`. So we are ignoring some errors which are not "file not found" errors.

Reply:
@cfergeau  The whole check is only to know if file is present of not and then try to process it. `!os.IsNotExist(err)` make sure all the other errors which are part of `os` package shouldn't need to handle one by one just to know if file is present or not.

```golang
package main

import (
	"os"
	"fmt"
)

func main() {
	 _, err := os.Stat("/etc/libvirt/libvirt.conf")
	if !os.IsNotExist(err) {
		fmt.Println("File Exist")
	}
	fmt.Println(err)
}
```

Following program tells that File exist even it doesn't have required permission and that all we want since we are deleting the file with sudo anyway, the way we created as part of the setup?

```
$ go run test.go 
File Exist
stat /etc/libvirt/libvirt.conf: permission denied
```

Reply:
if `os.Stat` fails with an error which is not "file not found", maybe we'll manage to delete the file, maybe we won't, we cannot know. So I'm only asking if we want to keep going and try to delete anyway and report an error only if deletion fails? Or do we want to return early reporting the error? 
All I'm saying is that it's not clear at all if the code is trying on purpose to delete the file when an error happened, or if it's an oversight, so I'm asking about it.